packages/runtime/lib/api-client.js

```diff
@@ -39,6 +39,23 @@ class RuntimeApiClient extends EventEmitter {
     })
   }
 
+  async setWorker (worker) {
+    this.worker.removeAllListeners('message')
```

Comment:
Can you add a check like that and use the `setWorker` in the constructor please.
```js
if (this.worker) {
  this.worker.removeAllListeners('message')
  this.#exitPromise.catch(() => {})
}
```